### PR TITLE
Add calendar popup for date/datetime fields

### PR DESCRIPTION
**Pull request by gacarrillor**

### CODE REVIEW

File: projectgenerator/libqgsprojectgen/generator/postgres/postgrescreator.py [hunk 1]
```diff
@@ -187,6 +188,17 @@ def layers(self):
...
+                if 'time' in fielddef['data_type'] or 'date' in fielddef['data_type']:
+                    field.widget = 'DateTime'
+                    field.widget_config['calendar_popup'] = True
+                    # timestamps don't need further configuration
+                    if 'time' in fielddef['data_type'] and not 'timestamp' in fielddef['data_type']:
+                        field.widget_config['field_format'] = 'HH:mm:ss'
+                        field.widget_config['display_format'] = 'HH:mm:ss'
+                    elif 'date' in fielddef['data_type']:
+                        field.widget_config['field_format'] = 'yyyy-MM-dd'
+                        field.widget_config['display_format'] = 'yyyy-MM-dd'
```
Comment (m-kuhn) [inline projectgenerator/libqgsprojectgen/generator/postgres/postgrescreator.py]: Is this format what's normally used at your place.
Here `dd.MM.yyyy` is more common. Could possibly determined from the QGIS locale, not sure.

Comment (m-kuhn) [inline projectgenerator/libqgsprojectgen/generator/postgres/postgrescreator.py]: Any feedback on this?

Comment (gacarrillor) [inline projectgenerator/libqgsprojectgen/generator/postgres/postgrescreator.py]: This definitely depends on the country. Perhaps we could format dates accordingly, but I'll need to have a look at how.

Comment (m-kuhn) [inline projectgenerator/libqgsprojectgen/generator/postgres/postgrescreator.py]: What's it like for you?

Comment (gacarrillor) [inline projectgenerator/libqgsprojectgen/generator/postgres/postgrescreator.py]: We can use 2017-08-24, but it's more common 24-08-2017. We never use 2017.08.24 nor 24.08.2017, although I like them :).

Comment (m-kuhn) [inline projectgenerator/libqgsprojectgen/generator/postgres/postgrescreator.py]: I think that would work nicely for me:

```py
>>> QLocale(QgsApplication.instance().locale()).dateFormat(QLocale.ShortFormat)
'dd.MM.yy'
>>> QLocale(QgsApplication.instance().locale()).dateTimeFormat(QLocale.ShortFormat)
'dd.MM.yy HH:mm'
```

What's it like for you?

Comment (gacarrillor) [inline projectgenerator/libqgsprojectgen/generator/postgres/postgrescreator.py]:
```
'd/MM/yy'
'd/MM/yy h:mm AP'
```
We can also use `/` :)
I think that's the way to go.

(I actually use English in my laptop, so for a user like me the date format would be a bit crazy 'M/d/yy', but I guess those are edge cases).

Comment (m-kuhn) [inline projectgenerator/libqgsprojectgen/generator/postgres/postgrescreator.py]: Actually, I'm also one of those edge cases...

Apparently it's only the default `en(_US)`

```py
>>> QLocale('en').dateFormat(QLocale.ShortFormat)
'M/d/yy'
>>> QLocale('en_GB').dateFormat(QLocale.ShortFormat)
'dd/MM/yyyy'
```

Let's do it anyway and blame it on Qt :P

Comment (gacarrillor) [reply projectgenerator/libqgsprojectgen/generator/postgres/postgrescreator.py]: Should be ready now.

**Pull request merged**